bagit.py

```diff
@@ -632,6 +645,23 @@ def _validate_bagittxt(self):
             if first_line.startswith(codecs.BOM_UTF8):
                 raise BagValidationError("bagit.txt must not contain a byte-order mark")
 
+    def _path_is_dangerous(self, path):
+        """
+        Return true if path looks dangerous, i.e. potentially operates
+        outside the bagging directory structure, e.g. ~/.bashrc, ../../../secrets.json,
+            \\?\c:\, D:\sys32\cmd.exe
+        """
+        # Disallow dots, slashes, and tildes as first character
+        disallow = r'.\/~'
```

Comment:
I think the approach of trying to blacklist known bad things is error prone – I would probably rewrite the regex as `if re.match(r'^[.\\/~]', path)` to make that clear but I'm wondering whether the better approach would be to use the Python `os.path` functions to fully expand the path and then confirm that it is still relative to the bag. I'd want to think carefully through the permutations but that would be something like calling `expanduser` and `abspath` on the input path, joining it with the bag base directory, and then testing that it still starts with the bag's base directory.